### ISSUE TITLE
Option to authorize new chats

### ISSUE THREAD

**Issue by Sepero** [oxen-io/session-android#665]
TLDR;
Feature request option to authorize/accept new chats from random people

====

As well as a feature request, this is a suggestion. To be more discriminatory about accepting new features. Normally I submit bugs, not feature requests, but I see features being implemented that I disagree with. I must explain.

First let me say, here are 2 recent features added.

* Sending a message in 1 to 1 chat the other user downloaded an image
* Hiding images in 1 to 1 chat until accepted

Both of these can be circumvented by one user inviting the other user to a Private Group of just the 2 of them.

In my humble opinion, these features do not add security nor value to Session, and should be removed.

For example, hiding images. Why? Because someone might send unwanted images (porn)? You won't know unless you accept viewing the images anyway. And if they do send unwanted images, then just block the person. Simple

The issue with both of these new features is with having contacts that can't be trusted. But you can't stop untrusted people from chatting with you on Session. And so the solution...

====

A better idea would be get rid of both those previous features, and add an option to manually authorize/accept new chats.

We can assume a scenario where Person X is trying to attack Person Y. Person X keeps generating new accounts, and sending garbage files and messages to Person Y.

Person Y could protect themselves by enabling "manually authorize new chats". Then anytime new people try to contact Person Y, those people can't get through unless specifically accepted/trusted. When people have trust in their contacts, then they can have worry free exchanges.

**Comment by Flam3z** [comment]
Its really good how telegram implements that in giving you the option to delete the chat, report it as spam or jist block the user all at once.

**Comment by hjubb** [comment]
This is a good idea, would definitely improve the experience for users potentially getting spammed if their session IDs are public or are active in open groups. I'm not sure reporting users could really be achieved though in @Flam3z comparison to Telegram but as adoption and the potential for abuse grows we'd definitely want to keep the experience as smooth for users as possible :+1:

**Comment by nielsandriesse** [comment]
Message requests are on our roadmap :)

**Comment by henk717** [comment]
In my opinion eyeberic's option 1 is the best way to go especially if we present more information to the user.
You can have a list of pending chat requests, this request can then be accompanied by 3 buttons, view, accept, block along with icons of the groups you have in common with this person so you know how relevant this request is to you (A user might be a moderator of some groups, while being sent unsolicited chats by other large groups). Choosing the view option opens the chat with the message box replaced for further acceptance or blocking.

In addition this will need to be synchronized across devices, currently blocks seem to be on a device per device basis making it hard to get rid of the spam. We would not want to frustrate users by having to either block or allow the conversations multiple times, so synchronizing the users actions across devices would be make this a much more pleasant experience.

**Comment by KeeJef** [comment]
Message requests have implemented this functionality and are now live, ID blinding limits spamming in open groups, and in the upcoming closed group release we will add message requests to closed groups too
